### PR TITLE
cleanup execution service

### PR DESCRIPTION
**Pull request by bhavesh-desai-scratch**
- Stop using inner subscriptions
- move openfin out of the execution service
- type the limit checker
- use race for timeout

### CODE REVIEW

File: src/client/src/services/executionService.ts [hunk 1]
```diff
@@ -32,80 +23,70 @@ const EXECUTION_REQUEST_TIMEOUT_MS = 30000
...
 
   executeTrade(executeTradeRequest: ExecuteTradeRequest) {
-    return new Observable<ExecuteTradeResponse>(obs => {
-      log.info('executing: ', executeTradeRequest)
-      const disposables = new Subscription()
+    return this.limitChecker(executeTradeRequest).pipe(
+      tap(() => log.info('executing: ', executeTradeRequest)),
+      take(1),
+      mergeMap(limitBreached => {
+        if (!limitBreached) {
```
Comment (fdore) [inline src/client/src/services/executionService.ts]: should that not be the opposite?

File: src/client/src/services/executionService.ts [hunk 2]
```diff
@@ -32,80 +23,70 @@ const EXECUTION_REQUEST_TIMEOUT_MS = 30000
 export default class ExecutionService {
   constructor(
     private readonly serviceClient: ServiceClient,
-    private readonly openFin: OpenFin
+    private readonly limitChecker: (
```
Comment (fdore) [inline src/client/src/services/executionService.ts]: 👍

File: src/client/src/services/executionService.ts [hunk 3]
```diff
@@ -32,80 +23,70 @@ const EXECUTION_REQUEST_TIMEOUT_MS = 30000
...
-          }
-          const request = this.serviceClient
-            .createRequestResponseOperation<
-              RawTradeReponse,
-              ExecuteTradeRequest
-            >(
-              ServiceConst.ExecutionServiceKey,
-              'executeTrade',
+        // After a longer period of time we know a trade is not coming back
+        const lastTimeout = timer(EXECUTION_REQUEST_TIMEOUT_MS).pipe(
```
Comment (fdore) [inline src/client/src/services/executionService.ts]: I would pbly share the timer...

Comment (fdore) [inline src/client/src/services/executionService.ts]: Would pbly share the timer..

Comment (bhavesh-desai-scratch) [reply src/client/src/services/executionService.ts]: Each request needs it own timer, right?

File: src/client/src/services/executionService.ts (hunk 1 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/services/executionService.ts]: Your right. Inverted the naming

File: src/client/src/services/executionService.ts (hunk 3 above)
Comment (bhavesh-desai-scratch) [reply src/client/src/services/executionService.ts]: Will leave in as discussed

**Pull request merged**